tutorials/networking/http_request_class.rst

```diff
@@ -1,70 +1,91 @@
-:article_outdated: True
-
 .. _doc_http_request_class:
 
 Making HTTP requests
 ====================
 
+Why use HTTP?
+-------------
+
+`HTTP requests <https://developer.mozilla.org/en-US/docs/Web/HTTP>`_ are useful
+to communicate with web servers and other non-Godot programs.
+
+Compared to Godot's other networking features (like
+:ref:`High-level multiplayer <doc_high_level_multiplayer>`),
+HTTP requests have more overhead and take more time to get going,
+so they aren't suited for real-time communication, and aren't great to send
+lots of small updates as is common for multiplayer gameplay.
+
+HTTP however offers interoperability with external
+web resources and is great at sending and receiving large amounts
+of data, for example to transfer files like game assets.
+
+So HTTP may be useful for your game's login system, lobby browser,
+to retrieve some information from the web or to download game assets.
+
+This tutorial assumes some familiarity with Godot and the Godot Editor.
+Refer to the :ref:`Introduction <toc-learn-introduction>` and the
+:ref:`Step by step <toc-learn-step_by_step>` tutorial, especially its
+:ref:`Nodes and Scenes <doc_nodes_and_scenes>` and
+:ref:`Creating your first script <doc_scripting_first_script>` pages if needed.
+
+HTTP requests in Godot
+----------------------
+
 The :ref:`HTTPRequest <class_HTTPRequest>` node is the easiest way to make HTTP requests in Godot.
-It is backed by the more low-level :ref:`HTTPClient <class_HTTPClient>`, for which a tutorial is available :ref:`here <doc_http_client_class>`.
+It is backed by the more low-level :ref:`HTTPClient <class_HTTPClient>`,
+for which a tutorial is available :ref:`here <doc_http_client_class>`.
 
-For the sake of this example, we will create a simple UI with a button, that when pressed will start the HTTP request to the specified URL.
+For this example, we will make a HTTP request to GitHub to retrieve the name
+of the latest Godot release.
 
 .. warning::
 
-    When exporting to Android, make sure to enable the ``INTERNET``
+    When exporting to **Android**, make sure to enable the **Internet**
     permission in the Android export preset before exporting the project or
     using one-click deploy. Otherwise, network communication of any kind will be
     blocked by Android.
 
-Preparing scene
----------------
-
-Create a new empty scene, add a CanvasLayer as the root node and add a script to it. Then add two child nodes to it: a Button and an HTTPRequest node. You will need to connect the following signals to the CanvasLayer script:
+Preparing the scene
+-------------------
 
-- Button.pressed: When the button is pressed, we will start the request.
-- HTTPRequest.request_completed: When the request is completed, we will get the requested data as an argument.
+Create a new empty scene, add a root :ref:`Node <class_Node>` and add a script to it.
+Then add a :ref:`HTTPRequest <class_HTTPRequest>` node as a child.
 
 .. image:: img/rest_api_scene.png
```

Comment:
The image still shows the CanvasLayer/Button setup. It should be updated to match the new description.

Reply:
Damn, you're a life saver - quite right, not sure how I missed this one in my local build 😀